File: 049_Group_Anagrams/solution.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        res = {}
        for w in strs:
            k = ''.join(sorted(w))
            if k not in res:
                res[k] = []
            res[k].append(w)
        return list(res.values())

File: 049_Group_Anagrams/solution.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> anagramGroups = new HashMap<>();

        for (String str : strs) {
            char[] charArray = str.toCharArray();
            Arrays.sort(charArray);
            String sortedKey = String.valueOf(charArray);
            anagramGroups.computeIfAbsent(sortedKey, key -> new ArrayList<>()).add(str);
        }
        return new ArrayList<>(anagramGroups.values());
    }
}
